contracts/zero-ex/contracts/src/features/MetaTransactionsFeature.sol

```diff
@@ -453,6 +457,81 @@ contract MetaTransactionsFeature is
         );
     }
 
+    /// @dev Execute a `INativeOrdersFeature.fillLimitOrder()` meta-transaction call
+    ///      by decoding the call args and translating the call to the internal
+    ///      `INativeOrdersFeature._fillLimitOrder()` variant, where we can override
+    ///      the taker address.
+    function _executeFillLimitOrderCall(ExecuteState memory state)
+        private
+        returns (bytes memory returnResult)
+    {
+        LibNativeOrder.LimitOrder memory order;
+        LibSignature.Signature memory signature;
+        uint128 takerTokenFillAmount;
+
+        bytes memory data = state.mtx.callData;
+        bytes memory args = new bytes(data.length - 4);
+        uint256 from;
+        uint256 to;
+        assembly {
```

Comment:
We can do that, but I'm not convinced it's going to be (much) better. We're already doing a low-level memory copy.

Reply:
Yeah I was thinking this could save us the `abi.decode`, which looking in Remix is ~4000 gas . Alternatively we could probably also just use `memCpy` to extract the `order`/`signature`.

Reply:
Actually given the time constraints I'm down to punt on optimizations for now. We can do more golfing in 4.1.